apps/application-system/form/src/routes/AssignApplication.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react'
+import { useHistory, useLocation } from 'react-router-dom'
+import { useMutation } from '@apollo/client'
+import qs from 'qs'
+
+import { Text, Page } from '@island.is/island-ui/core'
+import { NotFound } from '@island.is/application/ui-shell'
+
+import { ASSIGN_APPLICATION } from 'libs/application/graphql/src'
+
+export const AssignApplication = () => {
+  const location = useLocation()
+  const history = useHistory()
+  const queryParams = qs.parse(location.search, { ignoreQueryPrefix: true })
+  const [tokenParsingError, setTokenParsingError] = useState<Error | null>(null)
+
+  const [
+    assignApplication,
+    { loading, error: assignApplicationError },
+  ] = useMutation(ASSIGN_APPLICATION, {
+    onCompleted({ assignApplication }) {
+      history.push(`../application/${assignApplication.id}`)
+    },
+  })
+
+  const isMissingToken = !queryParams.token
+  const hasInvalidToken = tokenParsingError !== null
+  const couldNotAssignApplication = !!assignApplicationError
+
+  useEffect(() => {
+    if (isMissingToken) {
+      return
+    }
+
+    const { token } = queryParams
+    try {
+      setTokenParsingError(null)
+      assignApplication({
+        variables: {
+          input: {
+            token,
+          },
+        },
+      })
+    } catch (e) {
+      setTokenParsingError(e)
+    }
+  }, [])
```

Comment:
doesn't eslint complain that something is missing here?

Reply:
If you are referring to the empty dependency array, passing an empty dependency array to `useEffect` is the same as `componentDidMount`, will only run once. If you omit it then it will behave like `componentDidMount` and `componentDidUpdate`.

Reply:
I know how it works, I just thought that it would still complain :/ but if no complaints from eslint then I have no complaints :)